src/fastga/models/load_analysis/wing/aerostructural_loads.py

```diff
@@ -688,3 +707,89 @@ def add_point_mass(y_vector, chord_vector, point_mass_array, y_point_mass, point
         chord_vector_new = chord_vector
 
         return y_vector_new, chord_vector_new, point_mass_array_new
+
+
+def tank_volume_distribution(inputs, y_array_orig):
```

Comment:
I'm surprised to see this function copied here, after all the hard work to break it into differentiable openMDAO components. The cross-sections should be available from compute_wing_tank_cross_section.py (they seem promoted). Why then not use the variable here?

Reply:
That's a very good remark, I didn't catch it originally. However, solving it does not seem simple. The reason being that this input is only available when the "advanced" submodel is used to compute the MFW, meaning that if the "simple" one is used or if this model is used on its own, it will appear as a required input. And I don't think I want the user having to input that. 
The alternative solution would be to perform a check on the submodel used for the MFW computation and then if the simple one is used or if None is used, we would add the components.
Another alternative would be to not promote them either here or in the MFW computation so that we can put them in those two places.
What do you think ?

Reply:
I don't think I will have time to implement this properly as it would require a rework of the load computation module. Kept track of in #224 for further rework